manifests/machineconfig.crd.yaml

```diff
@@ -79,165 +136,186 @@ spec:
-                            type: boolean
-                          mask:
-                            description: "mask describes whether or not the service shall be masked. When true, the service is masked by symlinking it to /dev/null"
-                            type: boolean
                           contents:
-                            description: "contents is the contents of the unit"
+                            description: Contents is the contents of the unit
                             type: string
                           dropins:
-                            description: "dropins is the list of drop-ins for the unit"
+                            description: Dropins is the list of drop-ins for the unit
                             type: array
                             items:
-                              description: "items describes unit dropin"
+                              description: Items describes unit dropin
                               type: object
                               properties:
                                 contents:
-                                  description: "contents is the contents of the drop-in"
+                                  description: Contents is the contents of the drop-in
                                   type: string
                                 name:
-                                  description: "name is the name of the drop-in. This must be suffixed with '.conf'."
+                                  description: Name is the name of the drop-in. This must be suffixed
+                                    with '.conf'
                                   type: string
-          kargs:
-            description: "kargs contains a list of kernel arguments to be added"
-            type: array
-            items:
-              description: "kernel argument"
+                          enabled:
+                            description: Enabled describes whether or not the service shall be enabled.
+                              When true, the service is enabled. When false, the service is disabled.
+                              When omitted, the service is unmodified. In order for this to have any
+                              effect, the unit must have an install section
+                            type: boolean
+                          mask:
+                            description: Mask describes whether or not the service shall be masked.
+                              When true, the service is masked by symlinking it to /dev/null"
+                            type: boolean
+                          name:
+                            description: Name is the name of the unit. This must be suffixed with a
+                              valid unit type (e.g. 'thing.service')
+                            type: string
+            fips:
+              description: FIPS controls FIPS mode
+              type: boolean
+            kernelArguments:
```

Comment:
This should be `kargs` instead of kernelArguments.  From the [log](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/1485/pull-ci-openshift-machine-config-operator-master-e2e-aws/6547/artifacts/e2e-aws/pods/openshift-machine-config-operator_machine-config-controller-6bcffcc7fb-nmqdw_machine-config-controller.log) that would be reason why mcc pod is failing to render

Reply:
spoken a bit early, by looking at code don't think it's required. It could be another case of adding omitempty.